### PR TITLE
Remove rack

### PR DESCRIPTION
**Pull request by kbrock**
alt to #529 

I can't find a need for the `rack` dependency

### CODE REVIEW

File: webmock.gemspec [hunk 1]
```diff
@@ -18,6 +18,7 @@ Gem::Specification.new do |s|
   s.add_dependency 'addressable', '>= 2.3.6'
   s.add_dependency 'crack', '>=0.3.2'
   s.add_dependency 'hashdiff'
+  s.add_dependency 'rack'
```
Comment (bblimke) [inline webmock.gemspec]: is it needed here?

Comment (kbrock) [inline webmock.gemspec]: ugh

**Comment by kbrock** [comment]
sorry about that. thought I had gone back to master before creating this branch...

**Comment by bblimke** [comment]
no problem :) let's see what travis says

**Comment by davidbegin** [comment]
We need Rack for the test Rack app, but I'm not sure if we actually need it for anything else. I'm investigating now.

**Comment by bblimke** [comment]
@davidbegin which test rack app?

**Comment by kbrock** [comment]
other PR failed, I'll tweak that and push again.

**Comment by davidbegin** [comment]
@bblimke https://github.com/bblimke/webmock/blob/master/spec/support/my_rack_app.rb

**Comment by davidbegin** [comment]
I'm testing this out locally, push up my branch also https://travis-ci.org/bblimke/webmock/builds/85209158

**Comment by bblimke** [comment]
ahh Rack::Lock

**Comment by kbrock** [comment]
this needs to be a development dependency. I updated this accordingly. Test passed locally. (and this time I didn't have the other commit in there to fix)

**Comment by kbrock** [comment]
cool - you've got covered. just close if you want.

thanks for all the help.

**Comment by davidbegin** [comment]
Thanks for reporting this fast, and finding a fix, I going to work on getting a new version out.

**Pull request closed without merging**